src/ui/public/agg_types/controls/top.html

```diff
@@ -0,0 +1,29 @@
+<div class="form-group">
+  <label for="sort">
+    Sort On
+  </label>
+
+  <select
+    class="form-control"
+    name="sortField"
+    required
+    ng-model="agg.params.sortField"
+    ng-show="indexedFields.length"
+    auto-select-if-only-one="indexedFields"
+    ng-options="field as field.displayName group by field.type for field in indexedFields">
+  </select>
```

Comment:
Could you try to use the existing field input template here? https://github.com/Bargs/kibana/blob/70ba64fba2672e4c2937aa2ad0f588bc1f105263/src/ui/public/agg_types/controls/field.html